### PR TITLE
:book: Fix v1alpha4 references in clusterctl comments

### PR DESCRIPTION
**Pull request by killianmuldoon**
Fix up a couple of references to v1alpha4 in the clusterctl code.

Fixes #

### CODE REVIEW

File: cmd/clusterctl/api/v1alpha3/provider_type.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ type Provider struct {
 	// WatchedNamespace indicates the namespace where the provider controller is watching.
 	// If empty the provider controller is watching for objects in all namespaces.
 	//
-	// Deprecated: in clusterctl v1alpha4 all the providers watch all the namespaces; this field will be removed in a future version of this API
+	// Deprecated: in Cluster API v1alpha4 and above all providers watch all the namespaces; this field will be removed in a future version of this API
```
Comment (CecileRobertMichon) [inline cmd/clusterctl/api/v1alpha3/provider_type.go:53]: should this really say `Cluster API`? AFAIK clusterctl is still on v1alpha3

File: cmd/clusterctl/client/client.go [hunk 2]
```diff
@@ -57,6 +57,7 @@ type Client interface {
 	// PlanUpgrade returns a set of suggested Upgrade plans for the cluster, and more specifically:
 	// - Upgrade to the latest version in the v1alpha3 series: ....
```
Comment (sbueringer) [inline cmd/clusterctl/client/client.go:58]: I wonder how useful this list of contracts here is and if we'll be able to keep this up-to-date in the future (as we didn't in the past)

(similar in other places)

File: cmd/clusterctl/api/v1alpha3/provider_type.go (hunk 1 above)
Comment (killianmuldoon) [reply cmd/clusterctl/api/v1alpha3/provider_type.go:53]: This change is related to the CAPI contract version, not the clusterctl version which is why I updated this. AFAIU it's related to https://cluster-api.sigs.k8s.io/developer/providers/v0.3-to-v0.4.html#multi-tenancy

File: cmd/clusterctl/client/client.go (hunk 2 above)
Comment (killianmuldoon) [reply cmd/clusterctl/client/client.go:58]: Sure - probably better off just removing this.

File: cmd/clusterctl/api/v1alpha3/provider_type.go (hunk 1 above)
Comment (sbueringer) [reply cmd/clusterctl/api/v1alpha3/provider_type.go:53]: Looks like in our current code we always set this field to an empty string and never read it (except in an equals func, but I didn't find a usage for the equals func). So basically it seems to be dead code today.

I have no idea if the watch behavior was rather coupled to the clusterctl v1alpha3 API or the v1alpha3 contract.

My guess is, at this point there was no difference between:
* clusterctl v1alpha3
* contract v1alpha3
* other API types v1alpha3

I think technically there is no such thing as "all providers in Cluster API v1alpha4 and above". Cluster API itself doesn't have a global version which applies to core and all providers. The only thing that comes close is the contract version.

**Comment by sbueringer** [comment]
Overall lgtm from my side. No idea what we want to do with this: https://github.com/kubernetes-sigs/cluster-api/pull/8234#discussion_r1127156569

**Comment by killianmuldoon** [comment]
Update the comment on `WatchNamespace` to refer directly to the contract. We could, alternatively, remove the explanation if we're not confident that's the meaning of the original comment.

**Comment by sbueringer** [comment]
Fine for me!
lgtm pending regen

**Pull request merged**